packages/web-components/src/components/va-search/va-search.tsx

```diff
@@ -15,159 +16,348 @@ import {
+   * Tab key was added to aid in isListboxOpen state management.
+   */
+  private handleInputKeyDown = (event: KeyboardEvent) => {
+    const options = this.el.shadowRoot.querySelectorAll(
+      '[role="option"]',
+    ) as NodeListOf<HTMLLIElement>;
+    const firstOption = options[0];
+    const lastOption = options[options.length - 1];
+
+    switch (event.key) {
+      case 'ArrowDown':
+        if (!firstOption) return;
+        this.selectSuggestion(firstOption);
+        break;
+      case 'ArrowUp':
+        if (!lastOption) return;
+        this.selectSuggestion(lastOption);
+        break;
+      case 'Enter':
+        this.handleSubmit();
+        break;
+      case 'Escape':
+        this.inputRef.value = '';
+        break;
+      case 'Tab':
+        this.isListboxOpen = false;
+        break;
+      default:
+        break;
+    }
   };
 
-  private handleInputFocusEvent = (event: FocusEvent) => {
-    this.inputFocusEvent.emit(event);
+  // Button event handlers
+  /**
+   * Fires a submit event when search button is clicked
+   */
+  private handleButtonClick = () => {
+    this.handleSubmit();
   };
 
-  private handleInputKeyDownEvent = (event: KeyboardEvent) => {
-    this.inputKeyDownEvent.emit(event);
+  // Listbox event handlers
+  /**
+   * Sets search input value to the suggestion clicked,
+   * removes aria-selected from previously selected suggestion if it exists,
+   * closes the listbox and fires a submit event.
+   */
+  private handleListboxClick = (index: number) => {
+    const suggestion = this.el.shadowRoot.getElementById(
+      `listbox-option-${index}`,
+    );
+    if (!suggestion) return;
+    const selectedSuggestion = this.el.shadowRoot.querySelector(
+      '[aria-selected="true"]',
+    );
+    if (selectedSuggestion) selectedSuggestion.removeAttribute('aria-selected');
+    this.inputRef.value = suggestion.innerText;
+    this.inputRef.removeAttribute('aria-activedescendant');
+    this.isListboxOpen = false;
+    this.handleSubmit();
   };
 
-  // Button Event Handlers
-  private handleButtonClickEvent = (event: KeyboardEvent | MouseEvent) => {
-    this.buttonClickEvent.emit(event);
+  /**
+   * Implements keyboard interface from Keyboard Support and guidance from Role, Property, State, and Tabindex Attribute at
+   * https://www.w3.org/TR/wai-aria-practices-1.1/examples/combobox/aria1.1pattern/listbox-combo.html
+   *
+   * Enter key was modified to also fire a submit event.
+   */
+  private handleListboxKeyDown = (event: KeyboardEvent, index: number) => {
+    const options = this.el.shadowRoot.querySelectorAll(
+      '[role="option"]',
+    ) as NodeListOf<HTMLLIElement>;
+
+    switch (event.key) {
+      case 'ArrowUp':
+        const lastOption = options[options.length - 1];
+        if (index === 0) {
+          if (!lastOption) return;
+          this.selectSuggestion(lastOption);
+        } else {
+          if (!options[index - 1]) return;
```

Comment:
Similar question here. We know that `index` will be at least 1, so what kind of value in the `NodeList` would be falsey?